tests/sha1.cc

```diff
@@ -0,0 +1,92 @@
+/*
+ *    Copyright 2022 Cryspen Sarl
+ *
+ *    Licensed under the Apache License, Version 2.0 or MIT.
+ *    - http://www.apache.org/licenses/LICENSE-2.0
+ *    - http://opensource.org/licenses/MIT
+ */
+
+#include <fstream>
+
+#include <gtest/gtest.h>
+#include <nlohmann/json.hpp>
+
+#include "Hacl_Streaming_SHA1.h"
+
+#include "config.h"
+#include "util.h"
+
+using json = nlohmann::json;
+
+#define bytes std::vector<uint8_t>
+
+typedef struct
+{
+  bytes msg;
+  bytes md;
+} TestCase;
+
+std::vector<TestCase>
+read_json(char* test_file)
+{
+
+  // Read JSON test vector
+  std::ifstream json_test_file(test_file);
+  nlohmann::json test_vectors;
+  json_test_file >> test_vectors;
+
+  std::vector<TestCase> tests_out;
+
+  // Read tests
+  for (auto& test : test_vectors.items()) {
+    auto test_value = test.value();
+    auto msg = from_hex(test_value["msg"]);
+    auto md = from_hex(test_value["md"]);
+    tests_out.push_back({ msg, md });
+  }
+
+  return tests_out;
+}
+
+class Sha1 : public ::testing::TestWithParam<TestCase>
+{};
+
+TEST_P(Sha1, KAT)
+{
+  auto test = GetParam();
+
+  bytes digest(test.md.size());
+
+  // Init
+  Hacl_Streaming_SHA2_state_sha2_224* state =
```

Comment:
Sha1 state?

Reply:
https://github.com/cryspen/hacl-packages/blob/main/src/Hacl_Streaming_SHA1.c#L43

Reply:
😔 right, looks like a bug. Can you file something in hacl-star or let Jonathan know so he can file something?

Reply:
I take care of it.